core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1843,47 +1725,47 @@ private void produceMessagesWithTimestamp(int messageCount, long startingTimesta
         }
     }
 
-    private void consumeMessages(AtomicInteger totalMessagesConsumed,
+    private int consumeMessages(AtomicInteger totalMessagesConsumed,
                                  int totalMessages,
                                  String groupId,
                                  int consumerNumber,
                                  int maxPolls,
-                                 boolean commit,
-                                 CompletableFuture<Integer> future) {
-        try (KafkaShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(), groupId)) {
-            shareConsumer.subscribe(Collections.singleton(tp.topic()));
-            consumeMessages(shareConsumer, totalMessagesConsumed, totalMessages, consumerNumber, maxPolls, commit, future);
-        } catch (Exception e) {
-            fail("Consumer " + consumerNumber + " failed with exception: " + e);
-        }
+                                 boolean commit) {
+        return assertDoesNotThrow(() -> {
+            try (KafkaShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(
+                    new ByteArrayDeserializer(), new ByteArrayDeserializer(), groupId)) {
+                shareConsumer.subscribe(Collections.singleton(tp.topic()));
+                return consumeMessages(shareConsumer, totalMessagesConsumed, totalMessages, consumerNumber, maxPolls, commit);
```

Comment:
Here as well, can we add the custom error message with consumer number as well.

Reply:
Sorry for the late reply, I added error message to all assertDoesNotThrow methods in this test. Thanks!